### DISCUSSION TITLE
Detect SecureBoot status

### DISCUSSION THREAD

**Discussion by we-mi** [ipxe/ipxe#433] [Ideas]
In the last few days I spend some time playing around with secure boot and successfully chainloaded ipxe.efi over shim with secure boot enabled.

I am now struggling to detect the secure boot status from within ipxe and I wonder if it's somehow possible to implement this (or maybe I am missing some config variable where this status is already saved).

I would try to implement this myself and do a pull request but I honestly don't think I am capable of doing such a task as I just have basic experience/knowledge in writing applications in C.

So if anybody could help...this would be great. Even pointing me at the right directions so I can learn and implement this by myself would help.

**Comment by NiKiZe** [comment]
Is that status even available?
One way could be to try and run a noop exit 0 EFI binary (unsigned) if that fails SB is required, if not then you do not need SB.

**Reply by we-mi** [reply]
It should be available, at least from what I've read.
The secure boot variable has multiple attributes set (`NON_VOLATILE | BOOTSERVICE_ACCESS | RUNTIME_ACCESS`).
I've got this information by executing `dmpstore -b` in an UEFI Shell.

The definition of these attributes is:
> The variable is available for read and write access in both the pre-boot
environment and the OS runtime environment and the contents are persistent
across system resets and power cycles. These types of variables are typically
used to share persistent information among pre-boot components and the
operating system.
> https://edk2-docs.gitbook.io/edk-ii-uefi-driver-writer-s-guide/5_uefi_services/readme.2/525_getvariable_and_setvariable

Executing a non-signed EFI binary is something I can actually try. This could be an easy workaround, but in the long term an in-built solution would be faster and maybe more reliable.

Thank you for the hint :)

**Comment by keithga** [comment]
It is possible to read SecureBoot status with this command in ipxe:
```
echo SecureBoot ( [01] is enabled / [00] is disabled ): ${efi/SecureBoot}
```
